src/Foundation/Site.php

```diff
@@ -25,7 +25,10 @@ public static function fromPaths(array $paths)
         date_default_timezone_set('UTC');
 
         if (! static::hasConfigFile($paths->base)) {
-            return new UninstalledSite($paths, $_SERVER['REQUEST_URI']);
+            return new UninstalledSite(
+                $paths,
+                Arr::get($_SERVER, 'REQUEST_URI')
```

Comment:
What do we need the value for at runtime for the installer? Since this is for the `UninstalledSite`, we could just provide a bogus value when `$_SERVER['REQUEST_URI']` isn't set?

I'd find that cleaner in this case.

Reply:
When reviewing the old code, it looks like the old version must have had `url` set to `null` in the CLI, so a bogus value should work fine and be less invasive?

Reply:
Hmm, generally I'm not the biggest fan of bogus values, but in this case, we could just put in the nightly site's literal url I suppose? And since we'd expect a url to be provided on CLI install, I suppose we wouldn't run into this issue for actual usage?

Reply:
I would literally put a bogus value there - something that signals that this is intentionally bogus, because irrelevant. `doesnotmatter.tld` or `bogus.url` maybe?

Reply:
A comment can help explain the choice, of course.

Reply:
Wait, in the core source code, or in the nightly site config?

Reply:
Core source code. When we pass a base URL to the `UninstalledSite` in `Site.php`.

Reply:
Ah alright I get what you're getting at.

Small followup that I should have thought of earlier: is there a reason why we require `url`, and only `url`?

Reply:
Good question. Haven't thought that true, I guess. :wink:

Reply:
Or actually... it's the one that's used far more than any of the other ones, but obviously the other ones must be used somewhere, too.

Reply:
Tbh, that's mostly why I preferred being able to pass in required keys per-site, or maybe even a flag. How do you feel about the mutation-based approach that @luceos  had in the original PR?

Reply:
Meh. At that point, we could easily do two classes again.